engine/runner/src/main/java/org/enso/runner/Main.java

```diff
@@ -1294,9 +1306,67 @@ private void println(String msg) {
     System.out.println(msg);
   }
 
-  private void launch(String[] args) {
+  private void launch(String[] args) throws IOException, InterruptedException, URISyntaxException {
     var options = buildOptions();
     var line = preprocessArguments(options, args);
+
+    if (line.hasOption(JVM_OPTION)) {
+      var jvm = line.getOptionValue(JVM_OPTION);
+      var current = System.getProperty("java.home");
+      if (current == null || !current.equals(jvm)) {
+        println("Running with JVM: " + current);
+        println("Requested JVM: " + jvm);
+        var loc = Main.class.getProtectionDomain().getCodeSource().getLocation();
+        println("Location: " + loc);
+        var commandAndArgs = new ArrayList<String>();
+        if (jvm == null) {
```

Comment:
The `JVM_OPTION` takes zero or one argument. If the argument is present, it should be a path to the GraalVM to use to execute Enso process. The code below uses `ProcessBuilder` to construct such process - more or less the code is a copy of what is in `bin/enso` and `bin/enso.bat` files.

When the `--jvm` option is used without additional argument, then we should somehow locate the GraalVM in the installation. CCing @4e6 and @radeusgd (as author of [enso launcher](https://github.com/enso-org/enso/blob/2638ba8d8e170ee3bdd83a34263c12020adfdced/docs/distribution/launcher-cli.md) which deals with similar issues) to guide this part of the implementation.